src/pcre2posix.c

```diff
@@ -210,6 +211,8 @@ else
   ret = (int)len;
   }
 
+PCRE2_ASSERT(len > 0 || preg != NULL);
```

Comment:
Adding an assertion here lets Clang reason about when `preg` is NULL. There was no bug, but static analysis wasn't able to work out that checking `len` was equivalent to checking `preg != NULL`.